### PR TITLE
feat: suffix pruning

### PR DESCRIPTION
**Pull request by gk-kindred**

### CODE REVIEW

File: packages/talos_suffix/src/utils.rs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+/// Retrives all the items that are valid and not None.
+pub fn get_nonempty_suffix_items<'a, T: 'a>(items: impl Iterator<Item = &'a Option<T>> + 'a) -> impl Iterator<Item = &'a T> + 'a {
```
Comment (anil0906) [inline packages/talos_suffix/src/utils.rs:2]: Test can be written for these without the context of suffix

File: packages/talos_suffix/src/suffix.rs [hunk 2]
```diff
@@ -18,11 +21,29 @@ impl<T> Suffix<T>
...
 {
-    pub fn new(capacity: usize) -> Suffix<T> {
-        let suffix_vec: Vec<Option<SuffixItem<T>>> = vec![None; capacity + 1];
-        let messages = VecDeque::from(suffix_vec);
-
-        let meta = SuffixMeta { head: 0, prune_vers: None };
+    pub fn with_config(config: SuffixConfig) -> Suffix<T> {
+        let SuffixConfig {
+            capacity,
+            prune_start_threshold,
```
Comment (anil0906) [inline packages/talos_suffix/src/suffix.rs:27]: I think we can add docs on core components. Like meaning and use case of new variables. It becomes easier to understand the change.

**Review by anil0906 — APPROVED** [review]

File: packages/talos_suffix/src/utils.rs (hunk 1 above)
Comment (gk-kindred) [reply packages/talos_suffix/src/utils.rs:2]: updated in latest commit

File: packages/talos_suffix/src/suffix.rs (hunk 2 above)
Comment (gk-kindred) [reply packages/talos_suffix/src/suffix.rs:37]: Basic documentation added just for this method for now. Will add more documentation in subsequent PRs

**Pull request merged**